packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1578,29 +1594,34 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                     this.logger,
                 );
 
-                // Update stats to be reported in the peformance event.
-                gcStats.deletedGCNodes = deletedNodeIds.length;
-                gcStats.totalGCNodes = referencedNodeIds.length + gcStats.deletedGCNodes;
-
                 // Update our summarizer node's used routes. Updating used routes in summarizer node before
                 // summarizing is required and asserted by the the summarizer node. We are the root and are
                 // always referenced, so the used routes is only self-route (empty string).
                 this.summarizerNode.updateUsedRoutes([""]);
 
                 // Remove this node's route ("/") and notify data stores of routes that are used in it.
                 const usedRoutes = referencedNodeIds.filter((id: string) => { return id !== "/"; });
-                this.dataStores.updateUsedRoutes(usedRoutes);
+                const { dataStoreCount, unusedDataStoreCount } = this.dataStores.updateUsedRoutes(usedRoutes);
+
+                // Update stats to be reported in the peformance event.
+                gcStats = {
+                    deletedNodes: deletedNodeIds.length,
+                    totalNodes: referencedNodeIds.length + deletedNodeIds.length,
+                    deletedDataStores: unusedDataStoreCount,
+                    totalDataStores: dataStoreCount,
+                };
 
                 // If we are running in GC test mode, delete objects for unused routes. This enables testing scenarios
                 // involving access to deleted data.
                 if (this.gcTestMode) {
                     this.dataStores.deleteUnusedRoutes(deletedNodeIds);
                 }
             } catch (error) {
-                event.cancel(gcStats, error);
+                event.cancel(undefined, error);
```

Comment:
I think we can still pass gcStats here because we can get exception after calculating these and in that case recording these could be beneficial?